src/lib.rs

```diff
@@ -733,7 +736,17 @@ impl<T> IntoIterator for NonEmpty<T> {
 
 #[cfg(test)]
 mod tests {
+    #[cfg(feature = "serialize")]
```

Comment:
Perhaps having a submodule eg. `tests::serialize` with all the tests and custom types would be cleaner, it would only need a single `#[cfg(feature = "serialize")]`.

Reply:
also agreed.